src/main/resources/templates/fragments/inputs/number.html

```diff
@@ -6,17 +6,24 @@
       hasPattern=${!#strings.isEmpty(pattern)},
       hasLabel=${!#strings.isEmpty(label)},
       hasAriaLabel=${!#strings.isEmpty(ariaLabel)},
+      isRequiredInput=${requiredInputs.getOrDefault(inputName, false) || required != null && required},
```

Comment:
I would add parenthesis around the `required` part, like so `(required != null && required)` to make sure it's clear how to interpret this.  Just makes it a little easier to read and very clear that those work together.

Reply:
Done